actionpack/test/controller/action_pack_assertions_test.rb

```diff
@@ -470,6 +470,31 @@ def test_assert_redirection_with_status
     end
   end
 
+  def test_assert_redirection_with_status_outside_3XX_range
+    Rack::Utils::HTTP_STATUS_CODES.keys.each do |status|
+      if status < 300 || status > 399
+        assert_raise ArgumentError, match: "Expected :status to be a redirect, but was #{status}" do
+          assert_redirected_to("http://test.host/some/path", status: status)
+        end
+      end
+    end
+
+    ActionDispatch::Assertions::RESPONSE_PREDICATES.except(:redirect).keys.each do |status|
+      assert_raise ArgumentError, match: "Expected :status to be a redirect, but was #{status}" do
+        assert_redirected_to("http://test.host/some/path", status: status)
+      end
+    end
+
+    ActionDispatch::AssertionResponse::GENERIC_RESPONSE_CODES.except(:redirect).each do |status, pseudostatus|
+      assert_raise ArgumentError, match: "Expected :status to be a redirect, but was #{status}" do
+        assert_redirected_to("http://test.host/some/path", status: status)
+      end
+      assert_raise ArgumentError, match: "Expected :status to be a redirect, but was #{pseudostatus}" do
+        assert_redirected_to("http://test.host/some/path", status: pseudostatus)
+      end
+    end
```

Comment:
I think we can omit these assertions.  Based on

https://github.com/rails/rails/blob/1e4f16108401da9952bddf9333b99181f57fe631/actionpack/lib/action_dispatch/testing/assertions/response.rb#L33-L37

`assert_response` does not support arguments like "4XX", and `ActionDispatch::Assertions::RESPONSE_PREDICATES.keys` necessarily covers `ActionDispatch::AssertionResponse::GENERIC_RESPONSE_CODES.keys`.